pkg/phlare/modules.go

```diff
@@ -245,20 +246,18 @@ func (f *Phlare) setupWorkerTimeout() {
 }
 
 func (f *Phlare) initQuerier() (services.Service, error) {
-	var (
-		storeGatewayQuerier *querier.StoreGatewayQuerier
-		err                 error
-	)
-
-	// if a storage bucket is configure we need to create a store gateway querier
-	if f.storageBucket != nil {
-		storeGatewayQuerier, err = querier.NewStoreGatewayQuerier(f.Cfg.StoreGateway, nil, f.Overrides, log.With(f.logger, "component", "store-gateway-querier"), f.reg, f.auth)
-		if err != nil {
-			return nil, err
-		}
-	}
-
-	querierSvc, err := querier.New(f.Cfg.Querier, f.ring, nil, storeGatewayQuerier, f.reg, log.With(f.logger, "component", "querier"), f.auth)
+	newQuerierParams := &querier.NewQuerierParams{
+		Cfg:             f.Cfg.Querier,
+		StoreGatewayCfg: f.Cfg.StoreGateway,
+		Overrides:       f.Overrides,
+		CfgProvider:     f.Overrides,
+		StorageBucket:   f.storageBucket,
+		IngestersRing:   f.ring,
+		Reg:             f.reg,
+		Logger:          log.With(f.logger, "component", "querier"),
+		ClientOptions:   []connect.ClientOption{f.auth},
+	}
+	querierSvc, err := querier.New(newQuerierParams)
```

Comment:
There is no perfect solution for this. You mentioned most of the pros and cons of both approaches. 

Personally I find the code snippet you highlighted more readable compared to

`
querierSvc, err := querier.New(f.Cfg.Querier, f.ring, nil, storeGatewayQuerier, f.reg, log.With(f.logger, "component", "querier"), f.auth)
`

or:

```
querierSvc, err := querier.New(
    f.Cfg.Querier,
    f.ring,
    nil,
    storeGatewayQuerier,
    f.reg,
    log.With(f.logger, "component", "querier"),
    f.auth
)
```

I find the struct approach more readable because I can see what is being passed on without switching contexts and without paying attention to the argument order. The readability gains are higher when the number of arguments is higher.

As for safety, I expect constructor functions like this to perform validation on the input. E.g., we are already sending a `nil` in my last example which is maybe fine in this context but not in another and the function itself has to enforce this.

I am also open to any approach and would be curious to hear what others think.

@simonswine @bryanhuhta @korniltsev @petethepig any thoughts?